src/components/safe-apps/AppCard.tsx

```diff
@@ -90,6 +147,21 @@ const AppCard = ({ safeApp }: AppCardProps): ReactElement => {
             >
               <ShareIcon width={16} alt="Share icon" />
             </IconButton>
+            {onPin && (
+              <IconButton
+                aria-label={`${pinned ? 'Unpin' : 'Pin'} ${safeApp.name}`}
+                size="small"
+                onClick={(event) => {
+                  event.preventDefault()
+                  event.stopPropagation()
+                  onPin(safeApp.id)
```

Comment:
You could move this to it's own function. It's the same as line 102-104.

Reply:
Then it'd change the function signature for the `CompactAppCard` and become a little dirty, in my opinion. Duplication is cheaper than weak abstraction ser